cluster-commands.go

```diff
@@ -1247,6 +1247,61 @@ func (adm *AdminClient) SiteReplicationResyncOp(ctx context.Context, site PeerIn
 	return res, err
 }
 
+// PoolEndpoints represent endpoints in a given pool
+// along with its setCount and setDriveCount.
+type PoolEndpoints struct {
+	// indicates if endpoints are provided in non-ellipses style
+	Legacy       bool
+	SetCount     int
+	DrivesPerSet int
+	Endpoints    Endpoints
+	CmdLine      string
+	Platform     string
+}
+
+// EndpointServerPools - list of list of endpoints
+type EndpointServerPools []PoolEndpoints
+
+// Endpoints - list of same type of endpoint.
+type Endpoints []Endpoint
+
+// Endpoint - any type of endpoint.
+type Endpoint struct {
+	*url.URL
```

Comment:
How is that actually transported across the wire?

I would prefer a base type - presumably just a string.

Reply:
Will change to `string` as makes sense.

Reply:
Kept `url.URL` as non pointer as receiving side we need other fields for anonymizing the URL

Reply:
How is that sent in JSON?

Reply:
The endpoint object marshalled looks as below
```
{
    "Scheme" : "http",
    "Opaque" : "",
    "User" : null,
    "Host" : "127.0.0.1:9011",
    "Path" : "/tmp/xl1/16/disk1",
    "RawPath" : "",
    "OmitHost" : false,
    "ForceQuery" : false,
    "RawQuery" : "",
    "Fragment" : "",
    "RawFragment" : "",
    "isLocal" : false,
    "poolIdx" : 1,
     "setIdx" : 2,
     "diskIdx" : 5
}
```
the ones with camelCase are top level fields where I added `json:` tags but rest from url.URL and struct attribute names are used as is.